article/app/controllers/ArticleController.scala

```diff
@@ -160,3 +175,15 @@ object ArticleController extends Controller with RendersItemResponse with Loggin
   }
 
 }
+
+object LongCacheAction {
+  def apply[X](block: RequestHeader => Future[Result]) = {
```

Comment:
Is this covariant needed? I don't see `X` used anywhere, but I'm probably wrong.